providers/openmetrics/client_interceptor.go

```diff
@@ -0,0 +1,59 @@
+package metrics
+
+import (
+	openmetrics "github.com/prometheus/client_golang/prometheus"
+	"google.golang.org/grpc"
+
+	"github.com/grpc-ecosystem/go-grpc-middleware/v2/interceptors"
+)
+
+var (
+	// DefaultClientMetrics is the default instance of ClientMetrics. It is
+	// intended to be used in conjunction the default Prometheus metrics
+	// registry.
+	DefaultClientMetrics = NewClientMetrics()
+)
+
+func init() {
+	openmetrics.MustRegister(DefaultClientMetrics.clientStartedCounter)
+	openmetrics.MustRegister(DefaultClientMetrics.clientHandledCounter)
+	openmetrics.MustRegister(DefaultClientMetrics.clientStreamMsgReceived)
+	openmetrics.MustRegister(DefaultClientMetrics.clientStreamMsgSent)
+}
+
+// UnaryClientInterceptor is a gRPC client-side interceptor that provides Prometheus monitoring for Unary RPCs.
+func UnaryClientInterceptor() grpc.UnaryClientInterceptor {
+	return interceptors.UnaryClientInterceptor(&reportable{})
+}
+
+// StreamClientInterceptor is a gRPC client-side interceptor that provides Prometheus monitoring for Streaming RPCs.
+func StreamClientInterceptor() grpc.StreamClientInterceptor {
+	return interceptors.StreamClientInterceptor(&reportable{})
+}
+
+// EnableClientHandlingTimeHistogram turns on recording of handling time of
+// RPCs. Histogram metrics can be very expensive for Prometheus to retain and
+// query. This function acts on the DefaultClientMetrics variable and the
+// default Prometheus metrics registry.
+func EnableClientHandlingTimeHistogram(opts ...HistogramOption) {
+	DefaultClientMetrics.EnableClientHandlingTimeHistogram(opts...)
+	openmetrics.Register(DefaultClientMetrics.clientHandledHistogram)
```

Comment:
I think we should use `openmetrics.MustRegister` because the above function - `EnableClientHandlingTimeHistogram` is supposed to be called during server initialization, and not while scraping metrics. 

Thanks for pointing out this 😅 , would have been difficult to debug why this feature isn't working in case of error.

Reply:
ping @bwplotka for suggestions, although using `MustRegister` sounds fair to me :).

Reply:
Let's propagate the error to the user because the options might be used after the registry. So let the user handle the error, rather than panicking.